### ISSUE TITLE
No error for defaults with None, if they're not Optional

### ISSUE THREAD

**Issue by GoToCoding** [python/mypy#9228]
Look to this code:
```python
def sqr(x: int = None) -> Optional[int]:
    if x is not None:
        return x ** 2
    return None
```
mypy for this case should raise error, because of updates in https://www.python.org/dev/peps/pep-0484:

> A past version of this PEP allowed type checkers to assume an optional type when the default value is None, as in this code:
```pyython
def handle_employee(e: Employee = None): ...
```
> This would have been treated as equivalent to:
```python
def handle_employee(e: Optional[Employee] = None) -> None: ...
```
> This is no longer the recommended behavior. **Type checkers should move towards requiring the optional type to be made explicit**.

I didn't found any issues for this case, I'm sorry, if this issue is duplicate.
I checked with `mypy 0.782` that for code above there is no errors.

**Comment by JelleZijlstra** [comment]
This behavior is controlled by the `--no-implicit-optional` option (https://mypy.readthedocs.io/en/stable/command_line.html?highlight=implicit%20optional#cmdoption-mypy-no-implicit-optional). Mypy hasn't yet flipped the default for this option.

**Comment by GoToCoding** [comment]
Thank you for answer, I enabled given option.

**Comment by Akuli** [comment]
`--strict` enables `--no-implicit-optional` and a bunch of other flags
